larpandoracontent/LArHelpers/LArVertexHelper.cc

```diff
@@ -58,4 +59,19 @@ LArVertexHelper::ClusterDirection LArVertexHelper::GetClusterDirectionInZ(
     throw StatusCodeException(STATUS_CODE_FAILURE);
 }
 
+//-----------------------------------------------------------------------------------------------------------------------------------------
+
+bool LArVertexHelper::IsInFiducialVolume(const CartesianVector &vertex, const std::string &detector)
+{
+    if (detector == "dune_fd_hd")
+    {
+        const float x{vertex.GetX()};
+        const float y{vertex.GetY()};
+        const float z{vertex.GetZ()};
+        return -310 < x && x < 310 && -550 < y && y < 550 && 50 < z && z < 1244;
+    }
+    else
+        return false;
+}
+
```

Comment:
I'm guessing this is hard-coded because the fiducial volume cuts are less well-defined (and mostly exist in random analysis modules, rather than actual geometry level properties)?

Would it be difficult to swap to include a status code, or something here? Returning `false` right now means "not in fiducial volume or the detector string wasn't valid and a silent failure occurred", which could cause issues either for other detectors in the future, or just cause some annoying debugging later due to a dodgy typo in that string.

Reply:
Good points all round, I can update it to provide a more flexible return value (and it would be nice to have FV definition come from an upstream source in the future - I'm not aware of any such source right now).